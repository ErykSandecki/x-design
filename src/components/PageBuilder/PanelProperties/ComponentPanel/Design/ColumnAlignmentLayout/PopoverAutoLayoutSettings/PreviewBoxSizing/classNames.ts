// utils
import { composeClassNames } from 'utils';

export const className = 'PreviewBoxSizing';

export const classNames = composeClassNames(
  className,
  [className, 'excluded'] as const,
  ['tileLeft'] as const,
  ['tileRight'] as const,
  ['tileChildren'] as const,
  ['stroke', 'excluded'] as const,
);
