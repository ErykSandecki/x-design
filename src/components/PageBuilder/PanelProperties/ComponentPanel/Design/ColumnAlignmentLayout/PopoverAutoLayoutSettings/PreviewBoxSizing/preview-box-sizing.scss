.PreviewBoxSizing {
  $self: &;

  align-items: center;
  column-gap: 10px;
  display: flex;
  justify-content: center;
  position: relative;
  transform-origin: center center;
  transition: 100ms linear;

  &--excluded {
    transform: scale(1.2);
  }

  & * {
    transition: 100ms linear;
  }

  &__tile-left,
  &__tile-right {
    @include theme('neutral-3', background-color);

    align-items: center;
    border-radius: 2.5px;
    display: flex;
    height: 48px;
    justify-content: center;
  }

  &__tile-left {
    width: 48px;
  }

  &__tile-right {
    width: 92px;
  }

  &__tile-children {
    @include theme('neutral-5', background-color);

    height: 36px;
    width: calc(100% - 12px);
  }

  &__stroke {
    @include theme('neutral-2', border, '1px dashed');

    border-radius: 2.5px;
    height: 48px;
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 150px;

    &--excluded {
      height: calc(48px - 12px);
      width: calc(150px - 12px);
    }
  }
}
