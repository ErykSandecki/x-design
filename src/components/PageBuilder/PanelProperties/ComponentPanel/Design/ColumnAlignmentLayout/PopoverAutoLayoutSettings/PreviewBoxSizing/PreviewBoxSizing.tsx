import { FC } from 'react';

// components

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './preview-box-sizing.scss';

// types
import { Box, TPreviewData } from 'shared';

export type TPreviewBoxSizingProps = Pick<TPreviewData, 'activeOption'>;

const PreviewBoxSizing: FC<TPreviewBoxSizingProps> = ({ activeOption }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box
      classes={{
        className: cx(classNamesWithTheme[className].name, classNamesWithTheme[className].modificators[activeOption]),
      }}
    >
      <Box classes={{ className: cx(classNamesWithTheme.tileLeft) }}>
        <Box classes={{ className: cx(classNamesWithTheme.tileChildren) }} />
      </Box>
      <Box classes={{ className: cx(classNamesWithTheme.tileRight) }}>
        <Box classes={{ className: cx(classNamesWithTheme.tileChildren) }} />
      </Box>
      <Box
        classes={{
          className: cx(classNamesWithTheme.stroke.name, classNamesWithTheme.stroke.modificators[activeOption]),
        }}
      />
    </Box>
  );
};

export default PreviewBoxSizing;
