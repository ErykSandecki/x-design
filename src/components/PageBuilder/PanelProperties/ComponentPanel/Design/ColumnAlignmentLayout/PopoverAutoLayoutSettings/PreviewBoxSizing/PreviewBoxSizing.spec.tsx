// components
import PreviewBoxSizing from './PreviewBoxSizing';

// utils
import { customRender } from 'test';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('PopoverAutoLayoutSettings snapshots', () => {
  it('should render PreviewBoxSizing', () => {
    // before
    const { asFragment } = customRender(<PreviewBoxSizing activeOption="included" />);

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when excluded', () => {
    // before
    const { asFragment } = customRender(<PreviewBoxSizing activeOption="excluded" />);

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
