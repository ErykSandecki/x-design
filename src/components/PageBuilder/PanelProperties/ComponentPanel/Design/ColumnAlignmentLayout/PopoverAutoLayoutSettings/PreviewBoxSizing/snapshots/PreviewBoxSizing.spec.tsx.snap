// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverAutoLayoutSettings snapshots should render PreviewBoxSizing 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp PreviewBoxSizing PreviewBoxSizing--light"
    data-test-box=""
  >
    <div
      class="css-1jnxvbp PreviewBoxSizing__tileLeft PreviewBoxSizing__tileLeft--light"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp PreviewBoxSizing__tileChildren PreviewBoxSizing__tileChildren--light"
        data-test-box=""
      />
    </div>
    <div
      class="css-1jnxvbp PreviewBoxSizing__tileRight PreviewBoxSizing__tileRight--light"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp PreviewBoxSizing__tileChildren PreviewBoxSizing__tileChildren--light"
        data-test-box=""
      />
    </div>
    <div
      class="css-1jnxvbp PreviewBoxSizing__stroke PreviewBoxSizing__stroke--light"
      data-test-box=""
    />
  </div>
</DocumentFragment>
`;

exports[`PopoverAutoLayoutSettings snapshots should render when excluded 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp PreviewBoxSizing PreviewBoxSizing--light PreviewBoxSizing--excluded PreviewBoxSizing--excluded--light"
    data-test-box=""
  >
    <div
      class="css-1jnxvbp PreviewBoxSizing__tileLeft PreviewBoxSizing__tileLeft--light"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp PreviewBoxSizing__tileChildren PreviewBoxSizing__tileChildren--light"
        data-test-box=""
      />
    </div>
    <div
      class="css-1jnxvbp PreviewBoxSizing__tileRight PreviewBoxSizing__tileRight--light"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp PreviewBoxSizing__tileChildren PreviewBoxSizing__tileChildren--light"
        data-test-box=""
      />
    </div>
    <div
      class="css-1jnxvbp PreviewBoxSizing__stroke PreviewBoxSizing__stroke--light PreviewBoxSizing__stroke--excluded PreviewBoxSizing__stroke--excluded--light"
      data-test-box=""
    />
  </div>
</DocumentFragment>
`;
